var/spack/repos/builtin/packages/py-networkit/package.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyNetworkit(PythonPackage):
+    """NetworKit is a growing open-source toolkit for large-scale network analysis. Its aim is to provide tools for the analysis of large networks in the size range from thousands to billions of edges. For this purpose, it implements efficient graph algorithms, many of them parallel to utilize multicore architectures. These are meant to compute standard measures of network analysis, such as degree sequences, clustering coefficients, and centrality measures. In this respect, NetworKit is comparable to packages such as NetworkX, albeit with a focus on parallelism and scalability."""
+
+    homepage = "https://networkit.github.io/"
+    git      = "https://github.com/networkit/networkit.git"
+    version('6.0', tag='6.0', submodules=True)
```

Comment:
I personally prefer to download Python packages from PyPI, see https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#url